src/filters/filter_conversion.hpp

```diff
@@ -403,6 +404,259 @@ COMPILER_SUPRESS("-Weffc++")
     using filter_8bit_to_float = filter_to_float_template<uint8_t, filter_type::filter_8bit_to_float>;
     using filter_16bit_to_float = filter_to_float_template<uint16_t, filter_type::filter_16bit_to_float>;
 
+    template <typename T, filter_type own_type>
+    class float_map_range : public filter {
+    private:
+        double* input = nullptr;
+        double lower_bound_in = 0.0;
+        double upper_bound_in = 1.0;
+        double lower_bound_out = 0.0;
+        double upper_bound_out = 1.0;
+        bool limit = false;
+        T output = 0;
+    public:
+        float_map_range() : filter() {}
+        virtual ~float_map_range() {}
+
+        virtual void setup_filter(const std::map<std::string, std::string>& configuration, const std::map<std::string, std::string>& initial_parameters, const channel_mapping& input_channels, const std::string& own_id) override {
+            MARK_UNUSED(initial_parameters);
+//            MARK_UNUSED(configuration);
+            if(!input_channels.float_channels.contains("value_in")) {
+                throw filter_config_exception("Unable to link input of float map range filter conversion filter: channel mapping "
+                                              "does not contain channel 'value_in' of type 'float'.",
+                                              own_type, own_id);
+            }
+            this->input = input_channels.float_channels.at("value_in");
+
+            if constexpr (std::is_same<T, uint8_t>::value) {
+                this->upper_bound_out = 255.0;
+            } else if constexpr (std::is_same<T, uint16_t>::value) {
+                this->upper_bound_out = 65535.0;
+            }
```

Comment:
For the future: One can write this in a nicer way by creating a constexpr function and assigning the default value to the variable directly. That way the compiler has better abilities for optimization and the code looks a bit cleaner (IMHO).